docs/internal/github_actions.rst

```diff
@@ -0,0 +1,195 @@
+which can then up uploaded and shared between jobs (and the user)
+as part of GitHub Actions' "artifact" API. A cache key comparison made using
+the relevant git commit SHA in order to determine if there is any need to rebuild
+the artifact, or if the existing cached artifact can be used instead.
+
+To make CI runs quicker, a seperate build is made for ACE_TAO, rather
+than building it at the same time as OpenDDS. This allows us to cache
+ACE_TAO relatively often due to the stability of the ace6tao2 branch.
+
+Job Types
+=========
+
+There are a number of job types that are contained in the file build_and_test.yml.
+Where possible, a configuration will contain 3 jobs. The first job that
+is run is the *ACE_TAO_*. This will create an artifact which is used later
+by the OpenDDS build. The second job is *build_*, which uses the previous
+*ACE_TAO_* job to configure and build OpenDDS. This job will then export
+an artifact to be used in the third step. The third step is the *test_*
+job, which runs the appropriate tests for the associated OpenDDS
+configuration.
+
+Certain builds do not follow this 3 step model. Safety Profile builds are done
+in one step due to the cross-compile nature causing problems. Static and Release
+builds have a large footprint and therefore cannot fit the entire test suite onto
+a Github Actions runner.  As a result, they only build and run a subset of the tests
+in their final jobs, but then have multiple final jobs to increase test coverage. These
+jobs are prefixed by: *compiler_* which runs the ``tests/DCPS/Compiler tests``, *unit_*
+which runs the unit tests located in ``tests/DCPS/UnitTest`` and ``tests/unit-tests``, and
+*messenger_* which runs the tests in ``tests/DCPS/Messenger`` and ``tests/DCPS/C++11/Messenger``.
+
+In addition to these builds, there are some builds which will not run the test suite in
+an effort to shorten the runtime of the continuous integration.  An exception to this is
+that all builds which are not safety, and have ownership profile enabled, will run the
+``tests/cmake`` tests. Test runs which only contain CMake tests are prefixed by ``cmake_``.
+
+Test Results
+============
+
+The tests are run using `autobuild <https://github.com/DOCGroup/autobuild>`_ which creates a number of output files
+that are turned into a GitHub artifact. This artifact is processed by the
+"Check Test Results" workflow which modifies the files with detailed summaries of the test runs.
+After all of the Check Test Results jobs are complete, the test results will be posted in either
+the build_and_test or lint workflows. It is random which one of the workflows the results will appear
+in, so be sure to check both.
+
+Artifacts
+=========
+
+Artifacts from the continuous integration run can be downloaded by clicking details
+on one of the Build & Test runs. There is a dropdown on the bar next to "Re-run jobs"
+called "Artifacts" which lists each artifact that can be downloaded.
```

Comment:
I don't see a dropdown "Artifacts".

Reply:
If you click the "checks" tab in a PR and then Select "Build & Test", or if you select one of the "Build & Test"  results from the bottom of the "details" page in a PR, it will appear near the number of lines changed in PR.  It won't show up by default when you click checks because it defaults to opening lint, which we don't create any artifacts in.

Reply:
Clayton, could you check if the "Artifacts" dropdown wont show until the jobs are completed? If that's true, we may add "When the jobs are completed, a dropdown "Artifacts" will appear ...".